### ISSUE TITLE
can you help me,thank you.

### ISSUE THREAD

**Issue by ryanchi** [chobie/php-protocolbuffers#12]
my proto file design as
message perm {
    ALL = 0;
    SYSTEM = 1;
    ARTICLE = 2;
    PIC = 3;
}

message Admin {
    required string name = 1;
    required string password = 2;
    repeate perm purview = 3;
}

but the generate php file serializeToString  the admin message is error，the binary string is different from the protobuf-c pack the admin message。and i use the binary string decode the Admin message ，the object before serialize and after decode is different . so i think when i repeated an enum type ，the serialize function may be not correct.

can you help me solve the problem.thank you very much.

**Comment by chobie** [comment]
Hi, ryanchi

First, I correct proto file like this. 

```
enum perm {
  ALL = 0;
  SYSTEM = 1;
  ARTICLE = 2;
  PIC = 3;
}

message Admin {
  required string name = 1;
  required string password = 2;
  repeated perm preview = 3;
}
```

And generate php proto file with `protoc --php_out=php --plugin=./protoc-gen-php -I ./ -I /usr/local/include admin.proto`

```
<?php
require "php/admin.proto.php";

$admin = new Admin(array(
  "name" => "Chobie",
  "password" => "**********",
  "preview" => array(perm::PIC, perm::ARTICLE)
));

file_put_contents("out", $admin->serializeToString());
```

Read the raw data from python `protoc --python_out=py -I ./ -I /usr/local/include admin.proto`

```
import admin_pb2
import sys

admin = admin_pb2.Admin()
f = open("out", "rb")
admin.ParseFromString(f.read())
f.close()

print "Name: " + admin.name
print "Password: " + admin.password
print "Preview: "
for permission in admin.preview:
  print permission
```

it works.

output file contains this raw data.

```
$ hexdump -C out
00000000  0a 06 43 68 6f 62 69 65  12 0a 2a 2a 2a 2a 2a 2a  |..Chobie..******|
00000010  2a 2a 2a 2a 1a 02 03 02                           |****....|
00000018
```

but something wrong. it contains packed repeated files... 
maybe I put some bug on `protoc-php-gen`. for now, could you correct admin.proto.php?

```
      $desc->addField(3, new ProtocolBuffersFieldDescriptor(array(
        "type"     => ProtocolBuffers::TYPE_ENUM,
        "name"     => "preview",
        "required" => false,
        "optional" => false,
        "repeated" => true,
        "packable" => true, // fix this as false. we don't want to use packed field
        "default"  => perm::ALL,
      )));
```

Probably this will solve your problem.

I'll fix protoc-gen-php soon.

Thanks,

**Comment by ryanchi** [comment]
hi,chobie

after i edit the php class file as you said ,that problem has been solve.

Looking forward to you solve the problem about the protoc-php-gen.

best regards.

**Comment by chobie** [comment]
Hi, I've fixed protoc-php-gen on 9a1e187bb6d8f3c4aaa2fce8d41c807ff8dd3c1c.

Thanks,
